app/controllers/users/two_factor_authentication_setup_controller.rb

```diff
@@ -60,14 +60,24 @@ def two_factor_options_presenter
       TwoFactorOptionsPresenter.new(
         user_agent: request.user_agent,
         user: current_user,
-        phishing_resistant_required: service_provider_mfa_policy.phishing_resistant_required?,
-        piv_cac_required: service_provider_mfa_policy.piv_cac_required?,
+        phishing_resistant_required: phishing_resistant?,
+        piv_cac_required: piv_cac_required?,
         show_skip_additional_mfa_link: show_skip_additional_mfa_link?,
         after_mfa_setup_path:,
         return_to_sp_cancel_path:,
       )
     end
 
+    def phishing_resistant?
+      service_provider_mfa_policy.phishing_resistant_required? &&
+        !mfa_context.phishing_resistant_configurations.present?
+    end
+
+    def piv_cac_required?
+      service_provider_mfa_policy.piv_cac_required? &&
+        !mfa_context.piv_cac_configurations.present?
+    end
```

Comment:
I kinda feel like it might be better for this logic to live inside the presenter class, since that helps keep the controller leaner, and we can more easily add tests for it there. We also already have `mfa_policy` inside `TwoFactorOptionsPresenter`, which we can use `phishing_resistant_mfa_enabled?` from.

`MfaPolicy` doesn't have an equivalent to checking if PIV/CAC configuration exists, but maybe we could add a method, or create `mfa_context` inside `TwoFactorOptionsPresenter`.

Reply:
hmm yea I can update to address,